src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository)
+    {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        if (!$this->thereIsADiscount($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $mostRecentLogEntry = $this->getMostRecentLogEntry($channelPricing);
+
+        $logEntryWithTheLowestPriceInPeriod = $this
+            ->channelPricingLogEntryRepository
+            ->findOneWithTheLowestPriceInPeriod($mostRecentLogEntry, self::PERIOD_IN_DAYS)
+        ;
+
+        if ($logEntryWithTheLowestPriceInPeriod !== null) {
+            $channelPricing->setLowestPriceBeforeDiscount($logEntryWithTheLowestPriceInPeriod->getPrice());
+
+            return;
+        }
+
+        $lowestPriceBeyondPeriod = $this
+            ->channelPricingLogEntryRepository
+            ->findTheMostRecentOneBeyondPeriod($mostRecentLogEntry, self::PERIOD_IN_DAYS)
+        ;
+
+        if ($lowestPriceBeyondPeriod === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        if ($lowestPriceBeyondPeriod->getOriginalPrice() === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $channelPricing->setLowestPriceBeforeDiscount($lowestPriceBeyondPeriod->getPrice());
+    }
+
+    private function thereIsADiscount(ChannelPricingInterface $channelPricing): bool
+    {
+        return
+            $channelPricing->getOriginalPrice() !== null &&
+            $channelPricing->getPrice() !== $channelPricing->getOriginalPrice()
```

Comment:
```suggestion
            $channelPricing->getPrice() < $channelPricing->getOriginalPrice()
```
not sure about that but the discount is only when the price is less than the original price

Reply:
I set it on purpose, as we can currently have such a "special" promotion:
<img width="343" alt="image" src="https://user-images.githubusercontent.com/40125720/219010206-be376cca-086e-46df-8274-6279bde7ebb8.png">

Reply:
I'd treat that as a bug (which it is) and not base business logic on it.
On the off-chance that we want to acknowledge such a case as valid, then we'd need to add checking the `originalPrice` value as well since someone might've set it lower than the `price` and it was visible on the shop.